commit cacd6e73693eeaaed1dff1fc310c9ee1fd781c23
Author: Russell Sim <russell.sim@gmail.com>
Date:   Tue Sep 8 12:48:37 2020 +0200

    Remove -D argument from install

diff --git a/Makefile.in b/Makefile.in
index 688bc97113..4f1e2e65c7 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -765,7 +765,10 @@ install-etc:
 ### Install native compiled Lisp files.
 install-eln:
 ifeq ($(HAVE_NATIVE_COMP),yes)
-	find native-lisp -type f -exec ${INSTALL_DATA} -D "{}" "$(ELN_DESTDIR){}" \;
+	for file in $(shell find native-lisp -type f); do \
+	umask 022; ${MKDIR_P} $$(dirname $(ELN_DESTDIR)/$${file}); \
+	${INSTALL_DATA} "$${file}" "$(ELN_DESTDIR)$${file}"; \
+	done
 endif
 
 ### Build Emacs and install it, stripping binaries while installing them.
